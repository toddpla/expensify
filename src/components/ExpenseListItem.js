import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import 'numeral/locales/en-gb';

numeral.locale('en-gb');

const ExpenseListItem = ({history, dispatch, id, description, amount, createdAt}) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{numeral(amount / 100).format('$0,0')}
      -
      {moment(createdAt).format('MMMM Do YYYY')}</p>
  </div>
)

export default ExpenseListItem;
