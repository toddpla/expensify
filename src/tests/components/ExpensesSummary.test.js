import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';


test('should correctly render ExpenseSummary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expenseTotal={1123}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render ExpenseSummary with 0 expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expenseTotal={1123}/>);
  expect(wrapper).toMatchSnapshot();
});
