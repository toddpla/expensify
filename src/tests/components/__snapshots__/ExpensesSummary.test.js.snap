// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 0 expenses 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $11.23
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling 
    $11.23
  </h1>
</div>
`;
